src/classrooms/classrooms.service.ts

```diff
@@ -56,4 +56,36 @@ export class ClassroomsService {
       userId: user.id,
     });
   }
+
+  async getClassroomsForUser(userId: number): Promise<ClassroomResponseDto[]> {
+    const user = await this.em.findOne(User, { id: userId }, { populate: ["teacher"] });
+    if (!user) {
+      throw new NotFoundException("User not found");
+    }
+
+    if (!user.teacher) {
+      throw new ForbiddenException("Only teachers can view classrooms");
+    }
+
+    const classrooms = await this.em.find(
+      Classroom,
+      { teacher: user.teacher },
+      {
+        orderBy: { classTime: "DESC" },
+        populate: ["teacher"],
+      },
+    );
```

Comment:
Probably need to modify find queries based on role

Reply:
Bhaiya, I will do that after I work with the student enrollment table. For now I am only working with teacher creating a classroom.